src/main/java/uk/gov/justice/services/test/matchers/HasEventsMatcher.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.services.test.matchers;
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+
+public class HasEventsMatcher extends TypeSafeDiagnosingMatcher<List<JsonNode>>{
+
+    private final List<JsonNode> expectedEvents;
+
+    public HasEventsMatcher(final List<JsonNode> expectedEvents) {
+        this.expectedEvents = expectedEvents;
+    }
+
+    @Override
+    protected boolean matchesSafely(final List<JsonNode> actualEvents, final Description description) {
+
+        final int expectedEventsSize = expectedEvents.size();
+        final int actualEventsSize = actualEvents.size();
+
+        if(actualEventsSize >= expectedEventsSize) {
+            if (startsWith(expectedEvents, actualEvents)) {
+                return true;
+            }
+        }
+
+        if(actualEventsSize == 0) {
+            description.appendText("no events found in stream");
+        } else if(actualEventsSize == 1) {
+            description.appendText("found this event in stream:\n" + actualEvents.get(0));
```

Comment:
All output lines should use System.lineSeparator() rather than hard coding "\n" - check the other classes too including tests else they'll fail on Windows.